notanorm/ddl_helper.py

```diff
@@ -103,28 +103,43 @@ def __columns(self, ent) -> Tuple[Tuple[DbCol, ...], List[DbIndex]]:
         for col in ent.find_all(exp.Anonymous):
             if col.name.lower() == "primary key":
                 primary_list = [ent.name for ent in col.find_all(exp.Column)]
-                idxs.append(DbIndex(fields=tuple(primary_list), primary=True, unique=False))
+                idxs.append(DbIndex(fields=tuple(primary_list), field_lengths=(None,) * len(primary_list), primary=True, unique=False))
 
         for col in ent.find_all(exp.ColumnDef):
             dbcol, is_prim, is_uniq = self.__info_to_model(col)
             if is_prim:
-                idxs.append(DbIndex(fields=(col.name,), primary=True, unique=False))
+                idxs.append(DbIndex(fields=(col.name,), field_lengths=(None,), primary=True, unique=False))
             elif is_uniq:
-                idxs.append(DbIndex(fields=(col.name,), primary=False, unique=True))
+                idxs.append(DbIndex(fields=(col.name,), field_lengths=(None,), primary=False, unique=True))
             cols.append(dbcol)
         return tuple(cols), idxs
 
     @staticmethod
-    def __info_to_index(index) -> Tuple[DbIndex, str]:
+    def __info_to_index(index: Expression, dialect: str) -> Tuple[DbIndex, str]:
         """Get a DbIndex and a table name, given a sqlglot parsed index"""
         primary = index.find(exp.PrimaryKeyColumnConstraint)
         unique = index.args.get("unique")
         tab = index.args["this"].args["table"]
         cols = index.args["this"].args["columns"]
         field_names = [ent.name for ent in cols.find_all(exp.Column)]
+        if dialect != "mysql":
```

Comment:
this looks unnecesary... the loop below should add ent.name/None as needed

Reply:
Refactored to share more code between MySQL and other dialects -- though some special-casing is needed because of MySQL's dialect peculiarities.